src/main/java/com/ita/if103java/ims/service/impl/ItemServiceImpl.java

```diff
@@ -88,46 +90,54 @@ private boolean isEnoughCapacityInWarehouse(SavedItemDto savedItemDto) {
     }
 
     @Override
-    public ItemDto addItem(ItemDto itemDto) {
-        itemDao.addItem(itemDtoMapper.convertItemDtoToItem(itemDto));
+    public ItemDto addItem(ItemDto itemDto, User user) {
+        itemDto.setAccountId(user.getAccountId());
+        itemDao.addItem(itemDtoMapper.toEntity(itemDto));
         return itemDto;
     }
 
     @Override
-    public SavedItemDto findSavedItemById(SavedItemDto savedItemDto) {
-        return savedItemDtoMapper.convertSavedItemToSavedItemDto(savedItemDao.findSavedItemById(savedItemDto.getId()));
+    public SavedItemDto findSavedItemById(Long id) {
+        return savedItemDtoMapper.toDto(savedItemDao.findSavedItemById(id));
     }
 
 
     @Override
-    public boolean softDelete(ItemDto itemDto) {
-        return itemDao.softDeleteItem(itemDto.getName());
+    public boolean softDelete(Long id) {
+        return itemDao.softDeleteItem(id);
     }
 
     @Override
-    public SavedItemDto findByItemDto(ItemDto itemDto) {
-        return savedItemDtoMapper.convertSavedItemToSavedItemDto(savedItemDao.findSavedItemById(itemDto.getId()));
+    public List<SavedItemDto> findByItemId(Long id) {
+        return savedItemDtoMapper.toDtoList(savedItemDao.findSavedItemByItemId(id));
     }
 
 
     @Override
-    public boolean moveItem(WarehouseLoadDto warehouseLoadDto, SavedItemDto savedItemDto) {
-        savedItemDto.setItemDto(itemDtoMapper.convertItemToItemDto(itemDao.findItemById(savedItemDto.getId())));
+    public boolean moveItem(SavedItemDto savedItemDto, Long id, User user, Associate associate) {
+        savedItemDto.setItemDto(itemDtoMapper.toDto(itemDao.findItemById(savedItemDto.getId())));
         if (isEnoughCapacityInWarehouse(savedItemDto)) {
-            return savedItemDao.updateSavedItem(warehouseLoadDto.getId(), savedItemDto.getId());
+            eventService.create(ItemEventUtil.createEvent(savedItemDto, user, EventName.ITEM_MOVED));
+            transactionDao.create(ItemTransactionUtil.createTransaction(savedItemDto, user, associate, TransactionType.MOVE));
+            eventService.create(ItemEventUtil.createEvent(savedItemDto, user, EventName.ITEM_MOVED));
+            return savedItemDao.updateSavedItem(id, savedItemDto.getId());
         }
+        eventService.create(ItemEventUtil.createEvent(savedItemDto, user, EventName.LOW_SPACE_IN_WAREHOUSE));
```

Comment:
**IMPORTANT: In order to not block your PR please do this in next one!**
This event should be logged when warehouse is loaded more than 90 %. So need to add additional condition